File: 12.programmers_day/main.py
def gregorian(y):
	if ((y % 4 == 0 and y % 100 != 0) or (y % 400 == 0)):
		return f"12.09.{y}"

	return f"13.09.{y}"

def julian(y):
	if y % 4 == 0:
		return f"12.09.{y}"

	return f"13.09.{y}"

def dayOfProgrammer(year):
    if year > 1918:
        return gregorian(year)
    elif year == 1918:
        return f"26.09.{year}"
    else:
        return julian(year)

input = 2017
result = dayOfProgrammer(input)

print(
	"Given this values:\n",
	input,
	"\nResult:",
	result,
)

File: 12.programmers_day/main.go
package main

import (
	"fmt"
)

func gregorian(y int32) string {
	if (y%4 == 0 && y%100 != 0) || (y%400 == 0) {
		return fmt.Sprintf("12.09.%d", y)
	}
	return fmt.Sprintf("13.09.%d", y)
}
func julian(y int32) string {
	if y%4 == 0 {
		return fmt.Sprintf("12.09.%d", y)
	}
	return fmt.Sprintf("13.09.%d", y)
}

func dayOfProgrammer(year int32) string {
	if year > 1918 {
		return gregorian(year)
	} else if year == 1918 {
		return fmt.Sprintf("26.09.%d", year)
	} else {
		return julian(year)
	}
}

func main() {
	var input int32 = 2017
	result := dayOfProgrammer(input)

	fmt.Println(
		"Given this values:\n",
		input,
		"\nResult:",
		result,
	)
}
